src/software/ai/hl/stp/tactic/shadow_enemy/simulated_shadow_enemy_tactic_test.cpp

```diff
@@ -0,0 +1,198 @@
+    Robot shadowee(1, Point(0, -2), Vector(0, 0), Angle::quarter(),
+                   AngularVelocity::zero(), Timestamp::fromSeconds(0));
+    Robot enemy(2, Point(0, 2), Vector(0, 0), Angle::threeQuarter(),
+                AngularVelocity::zero(), Timestamp::fromSeconds(0));
+
+
+
+    EnemyThreat enemy_threat{shadowee,     false, Angle::zero(), std::nullopt,
+                             std::nullopt, 1,     enemy};
+
+    addFriendlyRobots(
+        {RobotStateWithId{.id = 0, .robot_state = shadower.currentState()}});
+    addEnemyRobots({RobotStateWithId{.id = 1, .robot_state = shadowee.currentState()},
+                    RobotStateWithId{.id = 2, .robot_state = enemy.currentState()}});
+
+
+    setBallState(BallState(Point(3, 0), Vector(0, 0)));
+    auto tactic = std::make_shared<ShadowEnemyTactic>();
+    tactic->updateControlParams(enemy_threat, 1.5);
+    setTactic(tactic);
+    setRobotId(0);
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [this, tactic, shadowee](std::shared_ptr<World> world_ptr,
+                                 ValidationCoroutine::push_type& yield) {
+            // As the shadowee is located at (0,-2) and the ball is located at (3,0),
+            // we would like to block the pass with a shadow distance of 1.5
+            Vector pass       = Point(3, 0) - shadowee.position();
+            Point destination = shadowee.position() + pass.normalize(1.5);
+            robotAtPosition(0, world_ptr, destination, 0.01, yield);
+        }};
+
+    std::vector<ValidationFunction> non_terminating_validation_functions = {};
+
+    runTest(terminating_validation_functions, non_terminating_validation_functions,
+            Duration::fromSeconds(5));
+}
+
+TEST_F(SimulatedShadowEnemyTacticTest, test_block_net_then_steal_and_chip)
+{
+    Robot shadower(0, Point(-2, 0), Vector(0, 0), Angle::zero(), AngularVelocity::zero(),
+                   Timestamp::fromSeconds(0));
+    Robot shadowee(1, Point(0, -2), Vector(0, 0), Angle::fromDegrees(135),
+                   AngularVelocity::zero(), Timestamp::fromSeconds(0));
+    Robot enemy(2, Point(0, 2), Vector(0, 0), Angle::threeQuarter(),
+                AngularVelocity::zero(), Timestamp::fromSeconds(0));
+
+
+    EnemyThreat enemy_threat{shadowee,     true, Angle::zero(), std::nullopt,
+                             std::nullopt, 1,    enemy};
+
+    addFriendlyRobots(
+        {RobotStateWithId{.id = 0, .robot_state = shadower.currentState()}});
+    addEnemyRobots({RobotStateWithId{.id = 1, .robot_state = shadowee.currentState()},
+                    RobotStateWithId{.id = 2, .robot_state = enemy.currentState()}});
+
+
+    setBallState(BallState(Point(0, -1.75), Vector(0, 0)));
+    auto tactic = std::make_shared<ShadowEnemyTactic>();
+    tactic->updateControlParams(enemy_threat, 2);
+    setTactic(tactic);
+    setRobotId(0);
+
+
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
```

Comment:
Just wondering, why does this need to be split into two sets of { } { }?

Reply:
It would also work if I put `ballKicked` in the first validation function, I just thought it would be easier to read, since one validation function is checking if the robot is at a position and the other validation function is checking if the ball is kicked.

Reply:
Makes sense, thanks for clarifying!